spec/cloud_test.go

```diff
@@ -218,4 +225,97 @@ func TestSuggestCloudGenerator(t *testing.T) {
 			t.Errorf("something went wrong")
 		}
 	}()
+
+	func() { // suggest AzureVMGenerator
+		ts := httptest.NewServer(http.HandlerFunc(func(res http.ResponseWriter, req *http.Request) {
+			if req.Header.Get("Metadata") != "true" {
+				http.NotFound(res, req)
+			}
+			fmt.Fprint(res, "ok")
+		}))
+		defer ts.Close()
+		u, _ := url.Parse(ts.URL)
+		azureVMBaseURL = u
+		defer func() { azureVMBaseURL = unreachableURL }()
+
+		cGen = SuggestCloudGenerator(&conf)
+		if cGen == nil {
+			t.Errorf("cGen should not be nil.")
+		}
+
+		gen, ok := cGen.CloudMetaGenerator.(*AzureVMGenerator)
+		if !ok {
+			t.Errorf("cGen should be *AzureVMGenerator")
+		}
+		if gen.baseURL != azureVMBaseURL {
+			t.Errorf("something went wrong")
+		}
+	}()
+}
+
+func TestSuggestCloudGenerator_CloudPlatformSpecified(t *testing.T) {
+	// All Cloud meta URLs are unreachable
+	unreachableURL, _ := url.Parse("http://unreachable.localhost")
+	ec2BaseURL = unreachableURL
+	gceMetaURL = unreachableURL
+	azureVMBaseURL = unreachableURL
+
+	func() {
```

Comment:
[nit] Lexical scoping of blocks instead of function should work here.

Reply:
https://github.com/mackerelio/mackerel-agent/pull/447/commits/6d9f5d2a098ffaf4fb8907e95409eb967cce043c